### PR TITLE
GUI:  finish public api of arcade.gui.property

### PR DESCRIPTION
**Pull request by eruvanos**
properties provide instance and value to listener (keep support for simple callbacks)

### CODE REVIEW

File: arcade/gui/property.py [hunk 1]
```diff
@@ -18,21 +18,39 @@ class _Obs(Generic[P]):
     def __init__(self, value: P):
         self.value = value
         # This will keep any added listener even if it is not referenced anymore and would be garbage collected
-        self.listeners: Set[Callable[[], Any]] = set()
+        self.listeners: Set[Callable[[Any, Any], Any]] = set()
```
Comment (pushfoo) [inline arcade/gui/property.py]: Can we make this a generic somehow? As a warning, although I haven't used `ParamSpec` extensively, I know it can be limited in some ways if you use either older versions of the `typing_extensions` backports.

File: arcade/gui/property.py [hunk 2]
```diff
@@ -18,21 +18,39 @@ class _Obs(Generic[P]):
...
 
 
 class Property(Generic[P]):
     """
     An observable property which triggers observers when changed.
 
+        def log_change(instance, value):
```
Comment (pushfoo) [inline arcade/gui/property.py:30]: Both Paul Craven and I prefer the explicit `.. code-block:: python` style.
```suggestion
... code-block:: python

        def log_change(instance, value):
```

**Review by pushfoo — CHANGES_REQUESTED** [review]
My overall impression is good and I like callbacks. I have one change to suggest and a question.

File: arcade/gui/property.py (hunk 1 above)
Comment (eruvanos) [reply arcade/gui/property.py]: the first arg can be what ever instance the property is defined in, the second is the new value, which could be resolved by Property[P] i guess.

File: arcade/gui/property.py [hunk 3]
```diff
@@ -18,21 +18,41 @@ class _Obs(Generic[P]):
...
         # This will keep any added listener even if it is not referenced anymore and would be garbage collected
-        self.listeners: Set[Callable[[], Any]] = set()
+        self.listeners: Set[Callable[[Any, P], Any]] = set()
 
 
 class Property(Generic[P]):
     """
     An observable property which triggers observers when changed.
 
+... code-block:: python
```
Comment (pushfoo) [inline arcade/gui/property.py]: My bad, typo in my last comment. It should be a double `.`, not a triple. :sweat_smile: 
```suggestion
.. code-block:: python
```

**Review by pushfoo — CHANGES_REQUESTED** [review]

**Comment by pushfoo** [comment]
TL;DR: After fixing my suggestion typo, merge as-is if you or einarf think it's fine

I'm not yet sure on the type annotations for this. My instinct is that there is something we can do to polish it, but I may also be wrong. Getting it merged and working takes priority over perfect types, however. It's not breaking anything else as far as I can tell, and the stricter type checkers aren't something we're in full compliance with yet anyway.
